.github/workflows/cloud-tpu-ci-nightly.yml

```diff
@@ -24,9 +24,12 @@ jobs:
     strategy:
       fail-fast: false # don't cancel all jobs on failure
       matrix:
-        jaxlib-version: ["pypi_latest", "nightly"]
+        jaxlib-version: ["pypi_latest", "nightly", "nightly+oldest_supported_libtpu"]
         tpu-type: ["v3-8", "v4-8", "v5e-4"]
     name: "TPU test (jaxlib=${{ matrix.jaxlib-version }}, ${{ matrix.tpu-type }})"
+    env:
+      LIBTPU_VERSION_DATE: 20230905
```

Comment:
20230905 is too old. It needs to be after version 0.29 (https://github.com/openxla/xla/blob/main/xla/pjrt/c/CHANGELOG.md#029-sep-6-2023), and the earliest is 20230907. I think you can also try install and check the PJRT API version of the libtpu (through a jax built from source).